### PR TITLE
Prepare CMP for usage with CDN

### PR DESCRIPTION
**Pull request by ajantsch**
In order to reduce costs and improve resiliency, we want to serve the scripts of the CMP with a CDN.

https://tvinsight.atlassian.net/browse/TVI-2219

The following changes were necessary:
- only configuration values are used in script templates
- sampling done clientside in `cmpapi.js` [Code](https://github.com/rbredtech/consent-management-provider/pull/70/files#diff-0dc1df36da18e7d7bc89fc42b94412fecbd342c88957ba222b837e34943f7d64R123)
- handling of the technical cookie was entirely moved to `cmpapi.js` (pure client-side setting and evaluation) [Code](https://github.com/rbredtech/consent-management-provider/pull/70/files#diff-0dc1df36da18e7d7bc89fc42b94412fecbd342c88957ba222b837e34943f7d64R125)
- reading of consent cookie/localStorage moved to `cmpapi.js` [Code](https://github.com/rbredtech/consent-management-provider/pull/70/files#diff-0dc1df36da18e7d7bc89fc42b94412fecbd342c88957ba222b837e34943f7d64R138)

### CODE REVIEW

File: src/templates/cmpapi.js [hunk 1]
```diff
@@ -1,6 +1,49 @@
...
+    return null;
+  }
+
+  function setCookie(name, value) {
+    var maxAge = 60 * 60 * 24 * 365 * 2; // 2 years
+    var cookie = name + '=' + value + ';max-age=' + maxAge + ';path=/;domain=<%-COOKIE_DOMAIN%>';
+    document.cookie = cookie;
+  }
+
+  function readStorage(key) {
```
Comment (Mgutjahr) [inline src/templates/cmpapi.js:28]: would rename to 
```suggestion
  function readStorageOrCookie(key) {
```

File: src/templates/cmpapi.js [hunk 2]
```diff
@@ -1,6 +1,49 @@
...
+      value = localStorage.getItem(key);
+      if (value) {
+        return value;
+      }
+    }
+    value = getCookie(key);
+    return value;
+  }
+
+  function writeStorage(key, value) {
```
Comment (Mgutjahr) [inline src/templates/cmpapi.js:40]: would rename to 
```suggestion
  function writeStorageAndCookie(key, value) {
```

File: src/templates/cmpapi.js [hunk 3]
```diff
@@ -189,7 +258,8 @@
         var consentByVendorIdParam = parameter;
         image = document.createElement('img');
         image.src =
-          '<%-CONSENT_SERVER_PROTOCOL%>://<%-CONSENT_SERVER_HOST%>/<%-API_VERSION%>/set-consent?consentByVendorId=' +
+          window.location.protocol +
+          '//<%-CONSENT_SERVER_HOST%>/<%-API_VERSION%>/set-consent?consentByVendorId=' +
```
Comment (Mgutjahr) [inline src/templates/cmpapi.js:262]: I'd add a random parameter here (e.g. timestamp) to guarantee that request is not cached by CDN

File: src/templates/cmpapi.js [hunk 4]
```diff
@@ -164,7 +232,8 @@
 
         image = document.createElement('img');
         image.src =
-          '<%-CONSENT_SERVER_PROTOCOL%>://<%-CONSENT_SERVER_HOST%>/<%-API_VERSION%>/set-consent?consentByVendorId=' +
+          window.location.protocol +
+          '//<%-CONSENT_SERVER_HOST%>/<%-API_VERSION%>/set-consent?consentByVendorId=' +
           serializeConsentByVendorId(consentDecisionByVendorId) +
           (channelId !== '' ? '&channelId=' + channelId : '');
```
Comment (Mgutjahr) [inline src/templates/cmpapi.js:238]: The setting of consent is a bit inconsistent in my opinion .. the cookie is set when the image is retrieved but the local storage is updated immediately in the updateLocalStorageConsent() method on line 240. - Should this function not be called in the image.onload callback?

**Review by Mgutjahr — COMMENTED** [review]
Some comments. - One thing i'm missing generally is that no cache headers are set in the controller. - This would however be necessary to control the caching depending on the endpoint that is called

File: src/controller/resetOldConsent.ts [hunk 5]
```diff
@@ -14,6 +17,8 @@ export const resetOldConsentController = (req: Request, res: Response) => {
     consent: req.query?.consent === "1",
   };
 
+  console.log(cookie);
+
```
[the remark this answers is not in the document]
Comment (ajantsch) [reply src/controller/resetOldConsent.ts]: removed, was debug leftover

**Review by Mgutjahr — APPROVED** [review]

**Pull request merged**